src/SFA.DAS.EmployerAccounts.Web.UnitTests/Controllers/EmployerTeamControllerTests/WhenAgreementsToSign.cs

```diff
@@ -64,5 +64,24 @@ public void ThenTheProviderPermissionsDeniedViewIsReturned()
             Assert.IsNotNull(result);
             Assert.AreEqual("ProviderPermissionsDenied", (result.Model as dynamic).ViewName);
         }
+
+        [Test]
+        public void ThenTheVacancyPanelShowsPreAgreementView()
+        {
+            // Arrange
+            var model = new AccountDashboardViewModel();
+            model.PayeSchemeCount = 1;
+            model.AgreementsToSign = true;
+
+            model.AccountViewModel = new Model.Account();
+            model.AccountViewModel.Providers.Add(new Model.Provider());
+
+            //Act
+            var result = _controller.Row2Panel2(model) as PartialViewResult;
+
+            //Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual("PreAgreementRecruitment", (result.Model as dynamic).ViewName);
```

Comment:
Should cast model to `PanelViewModel<AccountDashboardViewModel>`, rather than `dynamic`.